s3_management/manage.py

```diff
@@ -104,6 +105,31 @@
 
 S3IndexType = TypeVar('S3IndexType', bound='S3Index')
 
+
+@dataclasses.dataclass(frozen=True)
+class S3Object:
+    key: str
+    checksum: str | None
+
+    def __str__(self):
+        return self.key
+
+    def __cmp__(self, other):
+        return self.key == other.key
+
+    def __lt__(self, other):
+        return self.key < other.key
+
+    def __le__(self, other):
+        return self.key <= other.key
+
+    def __gt__(self, other):
+        return self.key > other.key
+
+    def __ge__(self, other):
+        return self.key >= other.key
```

Comment:
We can squeeze this with the `@functools.total_ordering` decorator into 2 methods. Also I believe you meant to define `__eq__` instead of `__cmp__` which is a python2-only thing.
```suggestion
@dataclasses.dataclass(frozen=True)
@functools.total_ordering
class S3Object:
    key: str
    checksum: str | None

    def __str__(self):
        return self.key

    def __eq__(self, other):
        return self.key == other.key

    def __lt__(self, other):
        return self.key < other.key
```